### DISCUSSION TITLE
this.driver.getPlatform is not a function

### DISCUSSION THREAD

**Discussion by milesje** [mikro-orm/mikro-orm#5413] [General]
### Describe the bug

Error reported after upgrading to 6.1.12 from 5.9.X.

> 
> [Nest] 45391  - 04/03/2024, 2:32:09 PM   ERROR [ExceptionHandler] this.driver.getPlatform is not a function
> TypeError: this.driver.getPlatform is not a function
>     at new MikroORM (/home/jmiles/Code/Github/market-edge-ws-nestjs/node_modules/@mikro-orm/core/MikroORM.js:95:105)
>     at Function.init (/home/jmiles/Code/Github/market-edge-ws-nestjs/node_modules/@mikro-orm/core/MikroORM.js:35:21)
>     at Injector.instantiateClass (/home/jmiles/Code/Github/market-edge-ws-nestjs/node_modules/@nestjs/core/injector/injector.js:369:37)
>     at callback (/home/jmiles/Code/Github/market-edge-ws-nestjs/node_modules/@nestjs/core/injector/injector.js:65:34)
>     at Injector.resolveConstructorParams (/home/jmiles/Code/Github/market-edge-ws-nestjs/node_modules/@nestjs/core/injector/injector.js:144:24)
>     at Injector.loadInstance (/home/jmiles/Code/Github/market-edge-ws-nestjs/node_modules/@nestjs/core/injector/injector.js:70:13)
>     at Injector.loadProvider (/home/jmiles/Code/Github/market-edge-ws-nestjs/node_modules/@nestjs/core/injector/injector.js:97:9)
>     at /home/jmiles/Code/Github/market-edge-ws-nestjs/node_modules/@nestjs/core/injector/instance-loader.js:56:13
>     at async Promise.all (index 4)
>     at InstanceLoader.createInstancesOfProviders (/home/jmiles/Code/Github/market-edge-ws-nestjs/node_modules/@nestjs/core/injector/instance-loader.js:55:9)
>     at /home/jmiles/Code/Github/market-edge-ws-nestjs/node_modules/@nestjs/core/injector/instance-loader.js:40:13
>     at async Promise.all (index 5)
>     at InstanceLoader.createInstances (/home/jmiles/Code/Github/market-edge-ws-nestjs/node_modules/@nestjs/core/injector/instance-loader.js:39:9)
>     at InstanceLoader.createInstancesOfDependencies (/home/jmiles/Code/Github/market-edge-ws-nestjs/node_modules/@nestjs/core/injector/instance-loader.js:22:13)
>     at /home/jmiles/Code/Github/market-edge-ws-nestjs/node_modules/@nestjs/core/nest-factory.js:108:17
>     at Function.asyncRun (/home/jmiles/Code/Github/market-edge-ws-nestjs/node_modules/@nestjs/core/errors/exceptions-zone.js:22:13)

### Reproduction

NestJS package using .env for mikro-orm setup.

### What driver are you using?

@mikro-orm/postgresql

### MikroORM version

6.1.12

### Node.js version

v20.11.1

### Operating system

Linux (NixOS)

### Validations

- [X] Read the [Contributing Guidelines](https://github.com/mikro-orm/mikro-orm/blob/master/CONTRIBUTING.md).
- [X] Read the [docs](https://mikro-orm.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/mikro-orm/mikro-orm/issues) that reports the same bug to avoid creating a duplicate.
- [X] Check that this is a concrete bug. For Q&A open a [GitHub Discussion](https://github.com/mikro-orm/mikro-orm/discussions) or join our [Slack](https://join.slack.com/t/mikroorm/shared_invite/enQtNTM1ODYzMzM4MDk3LWM4ZDExMjU5ZDhmNjA2MmM3MWMwZmExNjhhNDdiYTMwNWM0MGY5ZTE3ZjkyZTMzOWExNDgyYmMzNDE1NDI5NjA).
- [X] The provided reproduction is a [minimal reproducible example](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

**Comment by milesje** [comment]
Here is an example repo
https://github.com/milesje/mikro-orm-6-error

**Reply by B4nan** [reply]
How does your `.env` file look like? In v6 its no longer automatically evaluated, its only read and only the env vars prefixed with `MIKRO_ORM_` are respected.

https://mikro-orm.io/docs/upgrading-v5-to-v6#env-files-are-no-longer-automatically-loaded

**Reply by B4nan** [reply]
Tried to reproduce this again without any luck, you should be getting a validation error when there is no driver set. Only way to not see this error and get to your stack trace is by explicitly passing a config object without the driver (which also means explicitly disabling the validation it does in the constructor). The driver can be picked via env vars (via `MIKRO_ORM_TYPE`), that also works fine on my end.

**Reply by milesje** [reply]
@B4nan here is an example of my .env file...

```
MIKRO_ORM_CLI_USE_TS_NODE=true
MIKRO_ORM_TYPE=postgresql
MIKRO_ORM_ENTITIES=dist/common/entities
MIKRO_ORM_ENTITIES_TS=src/common/entities
MIKRO_ORM_HOST=<MY_DB_HOST>
MIKRO_ORM_PORT=5432
MIKRO_ORM_USER=<MY_USER_NAME>
MIKRO_ORM_PASSWORD=<MY_PASSWORD>
MIKRO_ORM_DB_NAME=postgres
MIKRO_ORM_FORCE_UNDEFINED=true
MIKRO_ORM_FORCE_UTC_TIMEZONE=true
MIKRO_ORM_SCHEMA_GENERATOR_DISABLE_FOREIGN_KEYS=true
MIKRO_ORM_SEEDER_PATH=dist/database/seeders
MIKRO_ORM_SEEDER_PATH_TS=src/database/seeders
MIKRO_ORM_MIGRATIONS_PATH=dist/migrations
MIKRO_ORM_MIGRATIONS_PATH_TS = src/migrations
MIKRO_ORM_MIGRATIONS_DISABLE_FOREIGN_KEYS=tru
```

**Reply by milesje** [reply]
if you pull the above linked repo and put a .env file in the root of the project with the above values... and then run `npm run start:dev` you will get the following error

> 
> [Nest] 35937  - 04/04/2024, 10:25:57 AM     LOG [NestFactory] Starting Nest application...
> [Nest] 35937  - 04/04/2024, 10:25:57 AM     LOG [InstanceLoader] MikroOrmModule dependencies initialized +24ms
> [Nest] 35937  - 04/04/2024, 10:25:57 AM   ERROR [ExceptionHandler] this.driver.getPlatform is not a function
> TypeError: this.driver.getPlatform is not a function
>     at new MikroORM (/home/jmiles/Code/Github/mikro-orm-6-error/node_modules/@mikro-orm/core/MikroORM.js:95:105)
>     at Function.init (/home/jmiles/Code/Github/mikro-orm-6-error/node_modules/@mikro-orm/core/MikroORM.js:35:21)
>     at Injector.instantiateClass (/home/jmiles/Code/Github/mikro-orm-6-error/node_modules/@nestjs/core/injector/injector.js:369:37)
>     at callback (/home/jmiles/Code/Github/mikro-orm-6-error/node_modules/@nestjs/core/injector/injector.js:65:34)
>     at Injector.resolveConstructorParams (/home/jmiles/Code/Github/mikro-orm-6-error/node_modules/@nestjs/core/injector/injector.js:144:24)
>     at Injector.loadInstance (/home/jmiles/Code/Github/mikro-orm-6-error/node_modules/@nestjs/core/injector/injector.js:70:13)
>     at Injector.loadProvider (/home/jmiles/Code/Github/mikro-orm-6-error/node_modules/@nestjs/core/injector/injector.js:97:9)
>     at /home/jmiles/Code/Github/mikro-orm-6-error/node_modules/@nestjs/core/injector/instance-loader.js:56:13
>     at async Promise.all (index 4)
>     at InstanceLoader.createInstancesOfProviders (/home/jmiles/Code/Github/mikro-orm-6-error/node_modules/@nestjs/core/injector/instance-loader.js:55:9)
>

**Reply by B4nan** [reply]
Ok, I think I found it.

**Reply by B4nan** [reply]
So the problem is that the nestjs adapter is actually using deprecated way to initialize the ORM. Should be easy to fix I hope.

**Reply by B4nan** [reply]
Here you go 54a37d016755f989545145e43abb56771b1e19ba

**Reply by milesje** [reply]
@B4nan thanks a lot for looking at this and the quick fix!
